drivers/gpio/gpio_hi3861.c

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (c) 2023 Chen Xingyu <hi@xingrz.me>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT hisilicon_hi3861_gpio
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/drivers/pinctrl/pinctrl_hi3861.h>
+#include <zephyr/arch/common/sys_io.h>
+#include <zephyr/irq.h>
+
+#define GPIO_BASE DT_INST_REG_ADDR(0)
```

Comment:
Will this IP always be single-instance? If not, the driver needs to support multiple instances.

Reply:
In fact it's a simplified re-implementation of `gpio_dw.c`. It's Hi3861-specific because it uses the SoC's pinctrl driver to set up internal pull-up/pull-down. So, at least in Hi3861, it's single instance.